manager/orchestrator/restart/restart.go

```diff
@@ -190,6 +189,87 @@ func (r *Supervisor) Restart(ctx context.Context, tx store.Tx, cluster *api.Clus
 	return nil
 }
 
+// TaskRestartDelay calculates and returns the next delay based t's RestartPolicy
+// Returns nil on error parsing the RestartPolicy
+func (r *Supervisor) TaskRestartDelay(ctx context.Context, t *api.Task) *time.Duration {
```

Comment:
I think it would be slightly better to return `(time.Duration, error)`.

Reply:
Yeah, it's considered poor form in go to use pointers as a sentinel for error conditions.